src/vizier/services/metadata/metadata_server.go

```diff
@@ -235,7 +236,11 @@ func main() {
 	updateCh := make(chan *k8smeta.K8sResourceMessage)
 	mdh := k8smeta.NewHandler(updateCh, k8sMds, k8sMds, nc)
 
-	k8sMc, err := k8smeta.NewController(updateCh)
+	namespaces := []string{v1.NamespaceAll}
+	if viper.IsSet("METADATA_NAMESPACES") {
+		namespaces = strings.Split(viper.GetString("METADATA_NAMESPACES"), ",")
+	}
```

Comment:
Can you explicitly declare this as a pflag in `init` and use that instead? Then you can make it a `StringSlice`.